lib/plugins/features/movies/controller.js

```
@@ -2,9 +2,24 @@
 
 const Movie = require('../../../models/movie');
 
+const renameTitleToName = (obj) => {
+  obj.name = obj.title;
+  Reflect.deleteProperty(obj, 'title');
+};
+
 exports.create = (payload) => {
-  payload.name = payload.title;
-  Reflect.deleteProperty(payload, 'title');
+  renameTitleToName(payload);
   return new Movie().save(payload)
   .then((movie) => new Movie({ id: movie.id }).fetch());
 };
+
+exports.get = (query) => {
+  if (Object.keys(query).length > 0) {
+    if (query.title) {
+      renameTitleToName(query);
+    }
+    return new Movie(query).fetch();
+  } else {
+    return new Movie().fetchAll();
```

Thinking about this endpoint from a REST perspective, what do you think about throwing a 404 instead of returning all the movies?

---

Done! I've added .or([peers]) to the Joi validator so that the user must include a search parameter.